panther_analysis_tool/backend/public_api_client.py

```diff
@@ -232,20 +231,7 @@ def bulk_validate(self, params: BulkUploadParams) -> BulkUploadValidateStatusRes
             params = {"input": receipt_id}  # type: ignore
             res = self._potentially_supported_execute(query, variable_values=params)  # type: ignore
             result = res.data.get("validateBulkUploadStatus", {})  # type: ignore
-            status = result.get("status", "")
-            error = result.get("error", "")
-
-            response = BulkUploadValidateStatusResponse(
-                error=error,
-                status=status,
-                result=BulkUploadValidateResult.from_json(result.get("result")),
-            )
-
-            if status in ["FAILED", "COMPLETED"]:
-                return response
-
-            if status not in ["NOT_PROCESSED"]:
-                raise BackendError(f"unexpected status: {status}")
+            return BulkUploadValidateStatusResponse.from_json(data=result)
```

Comment:
Doesn't this break the loop when any result is returned, even `NOT_PROCESSED`?

Reply:
Oh my goodness yes I completely lost context here. Thanks!

Reply:
[c4efb70](https://github.com/panther-labs/panther_analysis_tool/pull/351/commits/c4efb70973d04c27358b2601b537890b80e1441a)